examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Title: Train an Semantic Segmentation Model on Pascal VOC 2012 using KerasCV
+Author: [tanzhenyu](https://github.com/tanzhenyu)
+Date created: 2022/10/25
+Last modified: 2022/10/25
+Description: Use KerasCV to train a DeepLabV3 on Pascal VOC 2012.
+"""
+
+import tensorflow as tf
+
+import keras_cv
+from keras_cv.datasets.pascal_voc.segmentation import load
+from keras_cv.models.segmentation.deeplab import DeepLabV3
+
+strategy = tf.distribute.MirroredStrategy()
+local_batch = 2
+global_batch = local_batch * strategy.num_replicas_in_sync
+
+train_ds = load(split="train", data_dir=None)
+eval_ds = load(split="eval", data_dir=None)
+
+resize_layer = tf.keras.layers.Resizing(512, 512)
+
+
+def proc_train_fn(examples):
+    image = examples.pop("image")
+    image = tf.cast(image, tf.float32)
+    image = resize_layer(image)
+    cls_seg = examples.pop("class_segmentation")
+    zeros = tf.zeros_like(cls_seg)
+    mask = tf.equal(cls_seg, 255)
+    cls_seg = tf.where(mask, zeros, cls_seg)
+    cls_seg = resize_layer(cls_seg)
+    cls_seg = tf.cast(cls_seg, tf.uint32)
+    return image, cls_seg
+
+
+train_ds = train_ds.map(proc_train_fn, num_parallel_calls=1)
+train_ds = train_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(global_batch, drop_remainder=True)
+)
+
+eval_ds = eval_ds.map(proc_train_fn, num_parallel_calls=1)
+eval_ds = eval_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(global_batch, drop_remainder=True)
+)
+
+
+def pad_fn(image, y_true):
+    if isinstance(image, tf.RaggedTensor):
+        image = image.to_tensor()
+    if isinstance(y_true, tf.RaggedTensor):
+        y_true = y_true.to_tensor()
+    return image, y_true
+
+
+train_ds = train_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
+
+with strategy.scope():
+    backbone = keras_cv.models.ResNet50V2(
+        include_rescaling=True, weights="imagenet/classification-v2", include_top=False
```

Comment:
(optional): use just `"imagenet"` to use the best ImageNet weights (in its current form it will pin to v2)

Reply:
Done. I assume if the weights is not found, error will be thrown?